halo2_proofs/tests/frontend_backend_split.rs

```diff
@@ -182,7 +182,7 @@ impl<F: Field + From<u64>, const WIDTH_FACTOR: usize> MyCircuit<F, WIDTH_FACTOR>
             let d = meta.query_fixed(d, Rotation::cur());
             let lhs = [one.clone(), a, b].map(|c| c * s_lookup.clone());
             let rhs = [one.clone(), d, c].map(|c| c * s_ltable.clone());
-            lhs.into_iter().zip(rhs).collect()
+            zip(lhs, rhs).chain([(s_lookup, s_ltable)]).collect()
```

Comment:
Previously this was:
`s_lookup * [1, a, b] in s_ltable * [1, d, c]` <->
`[s_lookup*1, s_lookup*a, s_lookup*b] in [s_ltable*1, s_ltable*d, s_ltable*c]`

Now it's:
`[s_lookup*1, s_lookup*a, s_lookup*b, s_lookup] in [s_ltable*1, s_ltable*d, s_ltable*c, s_ltable]`

Isn't the first pair `(input_expr[0], table_expr[0])` redundant with the last one `(input_expr[3], table_expr[3])`?

Reply:
The answer is no. (starting from my understanding) @ed255 

I understood the original constraints as lookup of `[1, 1], [a, d], [b, c]`.
In other words, `[1, 1]` pair is one part of whole lookup, NOT as the pair of `selector`s.
At first, I thought this is redundant to lookup constants `[1, 1]`, but didn't want to change them.
I just wanted to add the new updates to the circuit, not changing any of original code.
That's why I kept the original constraints, just add the extra pair of `selector`s at the end.

Reply:
> The answer is no. (starting from my understanding) @ed255
> 
> I understood the original constraints as lookup of `[1, 1], [a, d], [b, c]`. In other words, `[1, 1]` pair is one part of whole lookup, NOT as the pair of `selector`s. At first, I thought this is redundant to lookup constants `[1, 1]`, but didn't want to change them.

That would be the case if we had
```rust
let lhs = [one.clone(), a, b];
```
But we actually had
```rust
let lhs = [one.clone(), a, b].map(|c| c * s_lookup.clone());
```

So originally it was `[s_lookup*1, s_ltable*1], [s_lookup*a, s_ltable*d], [s_lookup*b, s_ltable*c]`

Or I'm missing something?

Reply:
No, you don't miss anything.
Originally, it was `[s_lookup*1, s_ltable*1], [s_lookup*a, s_ltable*d], [s_lookup*b, s_ltable*c]`.
Now, I add the `[s_lookup, s_ltable]` at the end.
Even though the expressions share the same value, their meanings are different.
I mean `(input_expr[0], table_expr[0]) != (input_expr[3], table_expr[3])` <-> `[s_lookup*1, s_ltable*1] != [s_lookup, s_ltable]`

Reply:
@ed255 
I believe I misunderstand the issue behind this `redundancy`.
Here, I add the patch to correct the mistake. https://github.com/privacy-scaling-explorations/halo2/pull/347/commits/cdb157ba753d5b9662717527d7ae6f71bf0a9bed